pkg/storage/sharding/label.go

```diff
@@ -100,3 +105,36 @@ func ParseShardIDLabelValue(val string) (index, shardCount uint64, _ error) {
 
 	return index - 1, count, nil
 }
+
+// ShardFunc is a simpler API to the above, when you don't need to avoid allocations.
+func ShardFunc(l labels.Labels) uint64 {
+	hash, _ := labelsXXHash(make([]byte, 0, 1024), l)
+	return hash
+}
+
+// labelsXXHash replicates historical hash computation from Prometheus circa 2022.
+// The buffer b can be recycled to avoid allocations.
+func labelsXXHash(b []byte, ls labels.Labels) (uint64, []byte) {
+	b = b[:0]
+
+	for i, v := range ls {
+		if len(b)+len(v.Name)+len(v.Value)+2 >= cap(b) {
+			// If labels entry is 1KB+ do not allocate whole entry.
+			h := xxhash.New()
+			_, _ = h.Write(b)
+			for _, v := range ls[i:] {
```

Comment:
Should we use `ls.Range(...)` here to avoid assuming that it's a slice?

Reply:
This is just a copy of `Labels.Hash()`.

Reply:
I understand, but there it's part of the `labels` package, and we're trying to remove that assumption.